libp2p/src/builder.rs

```diff
@@ -59,13 +153,40 @@ impl TcpBuilder<AsyncStd> {
     }
 }
 
-#[cfg(all(feature = "tokio", feature = "tcp"))]
-impl TcpBuilder<Tokio> {
-    pub fn with_tcp(self) -> RelayBuilder<Tokio, impl AuthenticatedMultiplexedTransport> {
+#[cfg(feature = "tokio")]
+impl TcpNoiseBuilder<Tokio, Tls> {
+    #[cfg(feature = "noise")]
+    pub fn with_noise(self) -> RelayBuilder<Tokio, impl AuthenticatedMultiplexedTransport> {
         RelayBuilder {
             transport: libp2p_tcp::tokio::Transport::new(Default::default())
                 .upgrade(libp2p_core::upgrade::Version::V1Lazy)
-                .authenticate(libp2p_noise::Config::new(&self.keypair).unwrap())
+                // TODO: Handle unwrap?
```

Comment:
I'd say we just return a `Result` from this builder function. It should still be pretty ergonomic together with `?` for the user.

Reply:
Done :+1: